### DISCUSSION TITLE
How does image generation using emitters work?

### DISCUSSION THREAD

**Discussion by hartmark** [open-webui/open-webui#4658] [General]
How do I consistently generate images?

I have trouble getting emitters to work reliably, I can get it to sometimes generate images automatically, but not consistently.

I have installed the tool to generate images.
https://openwebui.com/t/techelpr/image/

Then I have created a custom model.
I'm using **llava:7b-v1.6** as base model, I have tried others without any luck.
I have this as system prompt:
```
You should never respond with that you cannot generate images or anything like that.

Generates an image of the prompt if it is something that could be thought as a concept to generate an image.

If your response has a generated image also suffix your response with "Prompt: <actual prompt of generated image>".
```

![image](https://github.com/user-attachments/assets/1e182cbf-56f9-4438-9257-831d055efe94)

If I use the generate image button
![image](https://github.com/user-attachments/assets/11823bff-6489-4f73-ba72-874c03418831)

I get an image so the generator is working well.

**Comment by hartmark** [comment]
The tool is also enabled in the model.

![image](https://github.com/user-attachments/assets/9149ca36-bb12-4219-b121-6a80b3ec7f75)

**Comment by justinh-rahb** [comment]
**LLaVa** is a multi-modal model for recognizing image ***input***, it's not specialized for creating image generation prompts or using tools/functions, which is the only role your selected LLM model plays in this function. I would retry this with a model like `llama3.1:latest` or `mistral:latest` instead.

TL;DR: just because **LLaVa** is a _vision_ model does not mean that it can use tools or understands how to create image generation prompts.

**Comment by hartmark** [comment]
Alright, I'll try that. It still feels like a bit voodoo of I get an image or not. 

I would like to consistently be able to say if I want the image or not.

I'd also would like to have the LLM to be able to enhance the prompt because I suck at prompting.

My idea is that one would type in "cat" and it will enhance the prompt for image generation to "cute kitten yada yada..." And also print out what it used as a prompt and then perhaps do a follow-up prompt by "make the cat red and it's now in the forest"
